joblib/externals/loky/process_executor.py

```diff
@@ -180,6 +180,9 @@ def flag_as_broken(self, broken):
 _global_shutdown = False
 
 
+_debug_exit_lock_temp = threading.Lock()
```

Comment:
- [ ] :Find better wording for this lock
- [ ] This lock is only necessary for pypy ?
- [ ] maybe there's a cleaner approach where the conflict issue when joining the executor thread both in `PypyTerminationThread` and `_python_exit`` can be addressed ?